### PR TITLE
Fix element mod p hash and use enum name in hash

### PR DESCRIPTION
**Pull request by AddressXception**
### Issue
*Link your PR to an issue*

Fixes #___

### Description
hashes were hasing on the python enumeration object which is difficult to reproduce across languages

### Testing
run the tests

### Checklist
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ ] 🤔 **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] ✅ **DO** check open PR's to avoid duplicates.
- [ ] ✅ **DO** keep pull requests small so they can be easily reviewed.
- [ ] ✅ **DO** build locally before pushing.
- [ ] ✅ **DO** make sure tests pass.
- [ ] ✅ **DO** make sure any new changes are documented.
- [ ] ✅ **DO** make sure not to introduce any compiler warnings.
- [ ] ❌**AVOID** breaking the continuous integration build.
- [ ] ❌**AVOID** making significant changes to the overall architecture.

💚Thank you!

### CODE REVIEW

File: src/electionguard/group.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def to_hex(self) -> str:
         Converts from the element to the hex representation of bytes. This is preferable to directly
         accessing `elem`, whose representation might change.
         """
-        h = format(self.elem, "02x")
+        h = format(self.elem, "02X")
```
Comment (AddressXception) [inline src/electionguard/group.py:91]: missed this in the tracker hash PR

**Review by pamtaro — APPROVED** [review]
:shipit: 🚀

**Review by rkorsak — APPROVED** [review]

**Comment by AddressXception** [comment]
this PR may have an unintended side effect that we are currently investigating.  don't merge it yet.

**Comment by pamtaro** [comment]
Confirmed that this works and side-effects were due to incomplete application of changes on the python web-api side during testing 👍

**Pull request merged**